src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -71,16 +74,39 @@ public class HttpClientRequestMessage : DataServiceClientRequestMessage, ISendin
         /// <summary>
         /// Constructor for HttpClientRequestMessage.
         /// Initializes the <see cref="HttpClientRequestMessage"/> class.
+        /// </summary>
+        public HttpClientRequestMessage(DataServiceClientRequestMessageArgs args)
+            : this(args, null)
+        {
+        }
+
+        /// <summary>
+        /// Constructor for HttpClientRequestMessage that takes a HttpClient instance.
+        /// Initializes the <see cref="HttpClientRequestMessage"/> class.
         /// The args.ActualMethod is the actual method. In post tunneling situations method will be POST instead of the specified verb method.
         /// The args.method is the specified verb method
+        /// The httpClient is an instance of HttpClient that will be used in making requests.
         /// </summary>
-        /// </summary>
-        public HttpClientRequestMessage(DataServiceClientRequestMessageArgs args) 
+        public HttpClientRequestMessage(DataServiceClientRequestMessageArgs args, HttpClient client)
             : base(args.ActualMethod)
         {
             _messageStream = new MemoryStream();
-            _handler = new HttpClientHandler();
-            _client = new HttpClient(_handler, disposeHandler: true);
+            if (client == null)
+            {
+#if NETSTANDARD2_0
+                var serviceProvider = new ServiceCollection().AddHttpClient().BuildServiceProvider();
+                var httpClientFactory = serviceProvider.GetService<IHttpClientFactory>();
+                _client = httpClientFactory.CreateClient();

```

Comment:
Since this service provider is created anew for each request, does that also mean we're creating a new client factory for each request? Is this the intended usage of a http client factory?